client/lib/domains/email-forwarding/index.js

```diff
@@ -1,11 +1,49 @@
-/** @format */
 /**
  * External dependencies
  */
-import { mapValues } from 'lodash';
+import { mapValues, includes } from 'lodash';
 import emailValidator from 'email-validator';
 
-export function validateAllFields( fieldValues ) {
+/**
+ * Internal dependencies
+ */
+import { hasGSuite } from 'lib/domains/gsuite';
+import { type as domainTypes } from 'lib/domains/constants';
+
+/**
+ * Retrieves the first domain that is eligible for Email Forwarding either from the current selected site or the list of domains.
+ *
+ * @param {String} selectedDomainName - domain name for the site currently selected by the user
+ * @param {Array} domains - list of domain objects
+ * @returns {String} - Eligible domain name
+ */
+function getEligibleEmailForwardingDomain( selectedDomainName, domains = [] ) {
+	const eligibleDomains = getEmailForwardingSupportedDomains( domains );
+	if ( selectedDomainName ) {
+		eligibleDomains.forEach( function( domain ) {
+			return domain.name === selectedDomainName && selectedDomainName;
+		} );
+	}
+	return ( eligibleDomains.length && eligibleDomains[ 0 ].name ) || '';
+}
+
+/**
+ * Filters a list of domains by the domains that eligible for Email Forwarding
+ *
+ * @param {Array} domains - list of domain objects
+ * @returns {Array} - Array of Email Forwarding supported domans
+ */
+function getEmailForwardingSupportedDomains( domains ) {
+	return domains.filter( function( domain ) {
+		const domainHasGSuite = hasGSuite( domain );
+		const wpcomHosted =
+			includes( [ domainTypes.REGISTERED ], domain.type ) && domain.hasWpcomNameservers;
+		const mapped = includes( [ domainTypes.MAPPED ], domain.type );
```

Comment:
This one also needs WP.com nameservers, I believe, so we might make it simpler if the above includes handled both mapping/registration.

Reply:
For MAPPED domains the value hasWpcomNameservers is undefined.  Got hit by that bug over on the G Suite functions a while back.